File: ReverseList.java
import java.util.ArrayList;
class ReverseList {
    public static void main(String args[]){
        ListNode list = new ListNode(1);
        list.next = new ListNode(2);
        list.next.next = new ListNode(3);
        list.next.next.next = new ListNode(4);
        list.next.next.next.next = new ListNode(5);
        ListNode current = reverseList(list);
        while(current != null){
            System.out.print(current.val+" ");
            current = current.next;
        }
    }
    public static ListNode reverseList(ListNode head) {
        ArrayList<Integer> list = new ArrayList<Integer>();
        while(head != null){
            list.add(head.val);
            head = head.next;
        }
        ListNode dummy = new ListNode(0);
        ListNode curr = dummy;
        for(int i = list.size()-1;i>=0;i--){
            curr.next = new ListNode(list.get(i));
            curr = curr.next;
        }
        return dummy.next;
    }
}

File: ReverseList.py
class Node:
   def __init__(self,data):
      self.data = data
      self.next = None
    
def reverse_list(head):
      prev = None
      current = head
       
      while current:
         next_node = current.next
         current.next = prev
         prev = current
         current = next_node
      
      return prev
   
def display(head):
      while head:
         print(head.data,end="->")
         head = head.next
      print("None")
    
n1 = Node(1)
n2 = Node(2)
n3 = Node(3)
n1.next = n2
n2.next = n3

print("Original List")
display(n1)

print("Reversed List")
obj = reverse_list(n1)
display(obj)


